spec/oil-builtin-argparse.test.sh

```diff
@@ -2,34 +2,48 @@
 #
 # Some thoughts before writing code.  Hm can we do this entirely in user code, not as a builtin?
 #
-# I think it modifies OPT
+# The following is as close as possible to the python argparse which seems to work well
 
+#### Argparse boolsche option and positional
 
-#### Argparse Prototype
-
-hay define argparse
-
-# Oops, we're running into this problem ...
+#I assume the hay definitions are given.
+ArgSpec myspec {
+  arg -v --verbose { type = Bool }
+  # TODO: can position be implicit (in given order?) or is the order not maintained in hay?
+  arg firstPositional
+}
+var args = ['-v' 'argument']
+argparse (myspec, args, :opts)
 
-hay define argparse/flag
+= opts
+= args
+## STDOUT:
+(OrderedDict)   <'verbose': True, 'firstPositional': 'argument'> 
+# TODO: Should this be empty afterwards? Is it even possible with above call?
+(List)   []
+## END
 
-# This means we need expr.Type objects?  
+#### Argparse basic help message
+ArgSpec myspec {
+  description = '''
+     Reference Implementation
+  '''
+  prog = "program-name"
+  arg -v --verbose { type = Bool, help = "Verbose" }
+  arg firstPositional
+}
+var args = ['-h' 'posone']
```

Comment:
This might be missing a comma

Try

`test/spec.sh oil-builtin-argparse` to test it

It would be useful to see the errors

Reply:
I saw only now that `test/spec.sh` allows `-f`, otherwise it only shows "FAIL" 😄  Thanks for spotting the typo